### PR TITLE
Fix/m2 3047 library  add to builder refresh workspace roles

### PR DESCRIPTION
**Pull request by muhurov**

### CODE REVIEW

File: src/modules/Library/features/Cart/Cart.tsx [hunk 1]
```diff
@@ -25,6 +24,7 @@ import { StyledTablePagination } from '../AppletsCatalog/AppletsCatalog.styles';
 import { DEFAULT_APPLETS_PER_PAGE, DEFAULT_PAGE } from '../AppletsCatalog/AppletsCatalog.conts';
 import { getSearchIncludes, getAddToBuilderData, navigateToBuilder } from './Cart.utils';
 import { useClearCart } from './Cart.hooks';
+import { useWorkspaceList } from '../../hooks';
```
Comment (IhorNesterenko99) [inline src/modules/Library/features/Cart/Cart.tsx]:
```suggestion
import { useWorkspaceList } from 'modules/Library/hooks';
```

and pls move to top imports

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ import {
 import { getArrayFromApplets, getSteps } from './AddToBuilderPopup.utils';
 import { addToBuilderPopupSchema } from './AddToBuilderPopup.schema';
 import { StyledContainer } from './AddToBuilderPopup.styles';
+import { useWorkspaceList } from '../../../hooks';
```
Comment (IhorNesterenko99) [inline src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.tsx]:
```suggestion
import { useWorkspaceList } from 'modules/Library/hooks';
```

File: src/modules/Library/hooks/useWorkspaceList.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  };
+
+  const checkIfHasAccessToWorkspace = async (ownerId: string) => {
+    const workspaces = await getWorkspaceList(ownerId);
+
+    return workspaces?.length > 0;
+  };
+
+  useEffect(() => {
+    if (typeof isAuthorized === 'undefined' || isAuthorized === true) {
```
Comment (IhorNesterenko99) [inline src/modules/Library/hooks/useWorkspaceList.ts:68]: why isAuthorized === 'undefined' check?

Comment (muhurov) [reply src/modules/Library/hooks/useWorkspaceList.ts:68]: isAuthorized is used only in Cart component, so for the AddToBuilderPopup there is no isAuthorized flag at all

File: src/modules/Library/features/Cart/Cart.tsx (hunk 1 above)
Comment (muhurov) [reply src/modules/Library/features/Cart/Cart.tsx]: done

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.tsx (hunk 2 above)
Comment (muhurov) [reply src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.tsx]: done

**Review by alex-i-e — APPROVED** [review]

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**